packages/material-ui/src/Typography/Typography.js

```diff
@@ -200,6 +200,8 @@ function Typography(props) {
   return <Component className={className} {...other} />;
 }
 
+let warnOnce = false;
```

Comment:
prop-types already have a builtin "warn-once". They use the message as the cache key which is preferred because they get warnings for each component stack.

Reply:
The builtin warn once if for a tuple (property, value).

Reply:
I was wrong: It does not include the stack: https://github.com/facebook/prop-types/blob/master/checkPropTypes.js#L78

Point still stands though. It warns once per variant.